third-party/realdds/include/realdds/dds-device-server.h

```diff
@@ -66,15 +66,12 @@ class dds_device_server
     bool is_valid() const { return( nullptr != _notification_server.get() ); }
     bool operator!() const { return ! is_valid(); }
 
-    void start_streaming( const std::vector< std::pair < std::string, image_header > > & ); //< stream_name, header > pairs
-    void stop_streaming( const std::vector< std::string > & stream_to_close );
+    std::map< std::string, std::shared_ptr< dds_stream_server > > const & streams() const { return _stream_name_to_server; }
```

Comment:
`dds_stream_server` should be thread-proof

Reply:
Thread-proof? C'mon there'll be one object in the system responsible for it. You want me to start putting in mutexes now?

Reply:
If there is only one object responsible than it's OK. But generally you won't always know who is accessing your returned member. One can call `start_streaming` while another calls `close` or such other race condition

Reply:
I don't think it's something to worry about at this time.